src/_imagingft.c

```diff
@@ -426,41 +707,56 @@ text_layout_fallback(
     }
 #endif
     for (i = 0; font_getchar(string, i, &ch); i++) {
-        (*glyph_info)[i].index = FT_Get_Char_Index(self->face, ch);
-        error = FT_Load_Glyph(self->face, (*glyph_info)[i].index, load_flags);
-        if (error) {
-            geterror(error);
-            return 0;
-        }
-        glyph = self->face->glyph;
-        (*glyph_info)[i].x_offset = 0;
-        (*glyph_info)[i].y_offset = 0;
-        if (kerning && last_index && (*glyph_info)[i].index) {
-            FT_Vector delta;
-            if (FT_Get_Kerning(
-                    self->face,
-                    last_index,
-                    (*glyph_info)[i].index,
-                    ft_kerning_default,
-                    &delta) == 0) {
-                (*glyph_info)[i - 1].x_advance += PIXEL(delta.x);
-                (*glyph_info)[i - 1].y_advance += PIXEL(delta.y);
+        int found = 0;
+        for (j = 0; !found && j < family->font_count; j++) {
+            FT_Face face = family->faces[j];
+            (*glyph_info)[i].index = FT_Get_Char_Index(face, ch);
+            if ((*glyph_info)[i].index != 0) {
+                found = 1;
             }
-        }
+            /* prefer first font's missing glyph if no font support this codepoint */
+            if (j == 0 || found) {
+                (*glyph_info)[i].face = face;
+                error = FT_Load_Glyph(face, (*glyph_info)[i].index, load_flags);
+                if (error) {
+                    geterror(error);
+                    return 0;
+                }
+                glyph = face->glyph;
+                (*glyph_info)[i].x_offset = 0;
+                (*glyph_info)[i].y_offset = 0;
+
+                /* This has been broken and had no effect for many years now...
```

Comment:
Could you expand on this?

Reply:
https://github.com/python-pillow/Pillow/issues/5191#issuecomment-756286205